lib/ui/screens/sections/preview_section.dart

```diff
@@ -31,28 +36,124 @@ class PreviewSection extends StatelessWidget {
 
     final previewWidgetSize = screenHeight / 1.5;
 
-    final gradient = context.select<GradientViewModel, AbstractGradient>(
```

Comment:
yeah! Makes sense! Actually we could even use the new Records feature to only listem to what we what need.

Reply:
hat to change it to listem to `(gradient, gradientDirection)`, if we don't listem to `gradientDirection` it won't update the UI, since the `gradient` hash won't change.

Reply:
Great work! 

To improve this, the gradient direction should be a part of the identity so we don't have to use `(gradient, gradientDirection)` and just use `gradient` directly. 

I filed an issue (https://github.com/victoreronmosele/flutter_gradient_generator/issues/81) to implement this improvement later and to remove the duplicated gradient identity check across the gradient styles.